src/main/java/org/dependencytrack/resources/v1/ProjectResource.java

```diff
@@ -464,4 +487,64 @@ public Response cloneProject(CloneProjectRequest jsonRequest) {
+            value = "Returns a list of all children for a project",
+            response = Project.class,
+            responseContainer = "List",
+            responseHeaders = @ResponseHeader(name = TOTAL_COUNT_HEADER, response = Long.class, description = "The total number of projects")
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 401, message = "Unauthorized"),
+            @ApiResponse(code = 404, message = "The UUID of the project could not be found")
+    })
+    @PermissionRequired(Permissions.Constants.VIEW_PORTFOLIO)
+    public Response getChildrenProjects(@ApiParam(value = "The UUID of the project to get the children from", required = true)
+                                            @PathParam("uuid") String uuid,
+                                        @ApiParam(value = "Optionally excludes inactive projects from being returned", required = false)
+                                        @QueryParam("excludeInactive") boolean excludeInactive) {
+        try (QueryManager qm = new QueryManager(getAlpineRequest())) {
+            final Project project = qm.getObjectByUuid(Project.class, uuid);
+            if (project != null) {
+                final PaginatedResult result = qm.getChildrenProjects(project.getUuid(), true, excludeInactive);
+                if (qm.hasAccess(super.getPrincipal(), project)) {
+                    return Response.ok(result.getObjects()).header(TOTAL_COUNT_HEADER, result.getTotal()).build();
+                } else {
+                    return Response.status(Response.Status.FORBIDDEN).entity("Access to the specified project is forbidden").build();
+                }
+            } else {
+                return Response.status(Response.Status.NOT_FOUND).entity("The UUID of the project could not be found.").build();
+            }
+        }
+    }
+
+    @GET
+    @Path("/withoutDescendantsOf/{uuid}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(
+            value = "Returns a list of all projects without the descendants of the selected project",
+            response = Project.class,
+            responseContainer = "List",
+            responseHeaders = @ResponseHeader(name = TOTAL_COUNT_HEADER, response = Long.class, description = "The total number of projects")
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 404, message = "The UUID of the project could not be found")
+    })
+    @PermissionRequired(Permissions.Constants.VIEW_PORTFOLIO)
+    public Response getProjectsWithoutDescendantsOf(@ApiParam(value = "The UUID of the project which descendants will be excluded", required = true)
+                                @PathParam("uuid") String uuid) {
+        try (QueryManager qm = new QueryManager(getAlpineRequest())) {
+            final Project project = qm.getObjectByUuid(Project.class, uuid);
+            if (project != null) {
+                final PaginatedResult result = qm.getProjectsWithoutDescendantsOf(project);
```

Comment:
There should be a `qm.hasAccess` check here, even though the project itself is not returned by this endpoint.